src/base_media_func.h

```diff
@@ -70,7 +71,12 @@ bool BaseSet<T>::FillSetDetails(const IniFile &ini, const std::string &path, con
 	}
 
 	fetch_metadata("version");
-	this->version = atoi(item->value->c_str());
+	auto value = ParseInteger(*item->value);
+	if (!value.has_value()) {
+		Debug(grf, 0, "Base {}set detail loading: {} field is invalid: {}", BaseSet::SET_TYPE, name, *item->value);
```

Comment:
String parameters feel wrong.
Probably should be better to use `BaseSet<T>::SET_TYPE` like everywhere else in the function.
And `name` is `this->name` while I think it should be the field name ("version" in this case), or it should mention "version" in the message.

Reply:
done.